easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -472,47 +472,57 @@ def dump(self, fp):
         """
         eb_file = file(fp, "w")
 
-        def to_str(x):
-            """Return quoted version of x"""
-            if isinstance(x, basestring):
-                if '\n' in x or ('"' in x and "'" in x):
-                    return '"""%s"""' % x
-                elif "'" in x:
-                    return '"%s"' % x
-                else:
-                    return "'%s'" % x
-            else:
-                return "%s" % x
-
         # ordered groups of keys to obtain a nice looking easyconfig file
         grouped_keys = [
+            ['easyblock'],
             ['name', 'version', 'versionprefix', 'versionsuffix'],
             ['homepage', 'description'],
             ['toolchain', 'toolchainopts'],
-            ['source_urls', 'sources'],
+            ['sources', 'source_urls'],
             ['patches'],
             ['builddependencies', 'dependencies', 'hiddendependencies'],
+            ['osdependencies'],
+            ['preconfigopts', 'configopts'],
+            ['prebuildopts', 'buildopts'],
+            ['preinstallopts', 'installopts'],
             ['parallel', 'maxparallel'],
-            ['osdependencies']
         ]
 
+        last_keys = [
+            ['sanity_check_paths'],
+            ['moduleclass'],
+        ]
+
+        def check_and_print(keyset):
+            """
+            Internal function checking for default values
+            """
+            for group in keyset:
+                printed = False
+                for key1 in group:
+                    val = self._config[key1][0]
+                    for key2, [def_val, _, _] in DEFAULT_CONFIG.items():
```

Comment:
the original code is pretty dumb here, there's no need for another for loop at all:

```python
for key in group:
  val = self._config[key][0]
  if key in DEFAULT_CONFIG:  
    def_val = DEFAULT_CONFIG[key]
    if val != def_val:
      ...
```